File: survey_authoring/src/main/java/com/stackroute/survey_authoring/Models/InactiveSurvey.java
package com.stackroute.survey_authoring.Models;

import com.stackroute.survey_authoring.Enums.SurveyType;
import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.Indexed;

import java.io.Serializable;
import java.util.Date;
import java.util.UUID;

@Data
public class InactiveSurvey implements Serializable {

    @Id
    @Indexed
    private UUID id;
    private String title;
    private String[] tags;
    private String description;
    private String targetAudience;
    private SurveyType surveyType;
    private ISQuestions[] questions;
    @Indexed
    private UUID createdBy;
    private Date createdOn;
    private Date updatedOn;
    private Object uiSetting;
    private Long activeTime;
}

File: surveyaugur_webapp/webapp/src/app/home/surveys/model/inactiveSurvey.ts
import { UUID } from "angular2-uuid";
import { ISQuestions } from "./isquestion";
import { SurveyType } from "./surveyType";


export class InActiveSurvey{
    id: UUID=UUID;
     title: string;
     tags: string[];
     description: string;
     targetAudience: string; //open for all, 20-40 years age male, 20-40 years age female, rural mass, urban mass
     surveyType: SurveyType;// Sectioned, Flat
     questions: ISQuestions[]; // can be array of questions or array of sections holding the questions
     uisettings:Object ;// all the custom settings for the ui
     createdOn: Date;
      createdBy: UUID;
     updatedOn: Date;
     activeTime: number;
   constructor( id?: UUID,
     title?: string,
     tags?: string[],
     description?: string,
     targetAudience?: string, //open for all, 20-40 years age male, 20-40 years age female, rural mass, urban mass
     surveyType?: SurveyType, // Sectioned, Flat
     questions?: ISQuestions[], // can be array of questions or array of sections holding the questions
     uiSettings?:Object ,// all the custom settings for the ui
     createdOn?: Date,
      createdBy?: UUID,
     updatedOn?: Date,
     activeTime?: number){}
}
